mesonbuild/interpreter/interpreter.py

```diff
@@ -3153,27 +3154,37 @@ def validate_installable_file(fpath: Path) -> bool:
             # subproject files, as long as they are scheduled to be installed.
             if validate_installable_file(norm):
                 return
-        norm = Path(os.path.abspath(Path(srcdir, subdir, fname)))
-        if os.path.isdir(norm):
-            inputtype = 'directory'
-        else:
-            inputtype = 'file'
-        if InterpreterRuleRelaxation.ALLOW_BUILD_DIR_FILE_REFERENCES in self.relaxations and builddir in norm.parents:
-            return
-        if srcdir not in norm.parents:
-            # Grabbing files outside the source tree is ok.
-            # This is for vendor stuff like:
-            #
-            # /opt/vendorsdk/src/file_with_license_restrictions.c
-            return
-        project_root = Path(srcdir, self.root_subdir)
-        subproject_dir = project_root / self.subproject_dir
-        if norm == project_root:
+
+        def do_validate_within_subproject(norm):
```

Comment:
```suggestion
        def do_validate_within_subproject(norm: Path) -> None:
```

Admittedly you're changing it to take a string in the next commit, but that commit doesn't have type annotations either.